### ISSUE TITLE
spatial data with cellchat 

### ISSUE THREAD

**Issue by chrkuo** [sqjin/CellChat#662]
thank you so much for creating this tool 

quick question regarding utilization of cell chat with spatial data: 

https://htmlpreview.github.io/?https://github.com/sqjin/CellChat/blob/master/tutorial/CellChat_analysis_of_spatial_imaging_data.html

the cell-cell communication inferred in this example it seems like the starting object is a deconvoluted object with integration of scRNA seq data set is that correct? so the resolution is more than spot resolution right? 

If I deconvoluted with giotto or CARD - is it possible to use the same object to then use cell chat? 

wondering how to interpret a deconvoluted spatial data-set cellchat result 

Thank you

**Comment by sqjin** [comment]
@chrkuo What is the output of giotto or CARD? Are they the proportion/probability of different cell types within each spot?

**Comment by chrkuo** [comment]
> @chrkuo What is the output of giotto or CARD? Are they the proportion/probability of different cell types within each spot? 

Yes it's probability. But also in the example what does the data tell you. Is it spot level or it's cell cell

**Comment by sqjin** [comment]
@chrkuo In the tutorial, a cell group label is assigned to each spot based on the maximum probability of predicted cell types.  Then CellChat computes the average expression of ligands/receptors within each cell group.

**Comment by chrkuo** [comment]
> @chrkuo In the tutorial, a cell group label is assigned to each spot based on the maximum probability of predicted cell types.  Then CellChat computes the average expression of ligands/receptors within each cell group.  Does this make sense to you?

But within a spot that could be 1-10 cells even with the maximum probability there's a chance that the ligand receptor is coming from a different cell correct? Unless the assumption is that the maximum probability is the ground truth

**Comment by chrkuo** [comment]
> > @chrkuo In the tutorial, a cell group label is assigned to each spot based on the maximum probability of predicted cell types.  Then CellChat computes the average expression of ligands/receptors within each cell group.  Does this make sense to you?
> 
> But within a spot that could be 1-10 cells even with the maximum probability there's a chance that the ligand receptor is coming from a different cell correct? Unless the assumption is that the maximum probability is the ground truth 

Is it then possible to use other objects that's been deconvoluted?

**Comment by sqjin** [comment]
@chrkuo Your understanding is correct. I am preparing another version to consider the possibility of multiple cell types within each spot.  At this moment, you should use the strategy of maximum probability and you can do it using other objects.

**Comment by chrkuo** [comment]
@sqjin 

Thank you - that would be extremely helpful looking forward to that.

**Comment by abulislam** [comment]
Hi,

@sqjin 
@chrkuo 

Cellchat for visiam ST data requires a dataframe consisting of the cell labels.
We used CytoSPACE for cell annotation using external reference scRNAseq data. The CytoSPACE provides more than one cell type (different fraction) per spot ID.

SpotID | CellType
TCGCGTAGCAGTGTCC-1 | mast_cell
TCGCGTAGCAGTGTCC-1 | NK_cell

Now how to deal with this?

thanks
